src/Form/Form.js

```diff
@@ -309,6 +335,12 @@ class Form {
      * Resets our input scoring and starts from scratch
      */
     recategorizeAllInputs () {
+        // If the form mutates too much, disconnect to avoid performance issues
+        if (this.rescanCount >= MAX_FORM_RESCANS) {
+            this.mutObs.disconnect()
+            return
+        }
+        this.rescanCount++
```

Comment:
This is the critical part of the fix. In the previous implementation, the counter was only updated within the observer callback, but the `recategorizeAllInputs` could also be called from `addInput` coming from the scanner. That was bypassing the failsafes and causing it to spin out of control in certain edge cases. Now the count is incremented right within the method so it's never missed 💪🤞.